R/derive_extreme_records.R

```diff
@@ -333,10 +355,12 @@ derive_extreme_records <- function(dataset = NULL,
     new_obs <- new_add_obs
   }
 
-  new_obs <- process_set_values_to(
-    new_obs,
-    set_values_to = set_values_to
-  )
+  new_obs <- new_obs %>%
+    select(!!!keep_vars_source) %>%
```

Comment:
`select()` should be called after `process_set_values()`. Then `set_values_to` can use variables which are dropped at the end.

Reply:
If `select()` is called after `process_set_values_to()`, then a newly created variable in `set_values_to()` would be dropped, since they will not be included in the tidyselect expression `exprs(everything())`.

Reply:
Are you sure?
`select(everything())` should select all variables of the input dataset. The variables added by `process_set_values_to()` are in the input dataset and should be kept.

Reply:
It would select everything in the input dataset. But if a new variable is created in `set_values_to()`  - for example 'DTYPE' - then that new variable would be dropped since `everything()` would not include it.

Reply:
It works in this example:
```
> myd <- data.frame(USUBJID = c("1", "2"))
> myvars <- exprs(everything())
> myd %>% process_set_values_to(exprs(DTYPE = "DERIVED")) %>% select(!!!myvars)
  USUBJID   DTYPE
1       1 DERIVED
2       2 DERIVED
```
Did I miss something?

Reply:
Yes it works. I think I missed something earlier. I've updated this now by moving `select()` after `process_set_values_to()`.